### DISCUSSION TITLE
Error "Method POST not allowed" when using actions

### DISCUSSION THREAD

**Discussion by xpluscal** [sveltejs/kit#6744] [Q&A]
In my +page.server.ts file I am using actions like so:

```
export const actions:Action = {
  add: async ({ cookies, request, locals }) => {
    const form = await request.formData();
    let res = await api('POST', `/reso`, Object.fromEntries(form));
    console.log("FINAL RES: ", res);
    return {success:true}
  }
};
```

But when running my form with:

```
<form action="?/add" method="POST>...</form>
```

I  get a 405 with the error "Method POST not allowed".
What am I doing wrong?
Help much appreciated :)

**Comment by teemingc** [comment]
Can you share a bit more about your error and code? (e.g., what line the error occurs on)
